packages/yarnpkg-core/sources/WorkerPool.ts

```diff
@@ -1,53 +1,67 @@
 import {cpus}   from 'os';
-import PQueue   from 'p-queue';
+import PLimit   from 'p-limit';
 import {Worker} from 'worker_threads';
 
+const kTaskInfo = Symbol(`kTaskInfo`);
+
+type PoolWorker<TOut> = Worker & {
+  [kTaskInfo]: null | { resolve: (value: TOut) => void, reject: (reason?: any) => void };
+};
+
 export class WorkerPool<TIn, TOut> {
-  private pool: Array<Worker> = [];
-  private queue = new PQueue({
-    concurrency: Math.max(1, cpus().length),
-  });
+  private workers: Array<PoolWorker<TOut>> = [];
+  private limit = PLimit(Math.max(1, cpus().length));
 
   constructor(private source: string) {
-    const timeout = setTimeout(() => {
-      if (this.queue.size !== 0 || this.queue.pending !== 0)
-        return;
+    setInterval(() => {
```

Comment:
As is, this setInterval will keep executing even after all the processes have been killed (vs the previous setTimeout-based implementation, which stops once there's nothing to cleanup).

Reply:
Yeah, since it's `unref`ed I didn't think much of it but I agree, it should clean up after itself 👍